packages/@azure/servicebus/data-plane/test/testUtils.ts

```diff
@@ -39,104 +42,323 @@ export enum ClientType {
+        env.resourceGroup,
+        env.servicebusNamespace,
+        queueName,
+        parameters,
+        function(error: any): void {
+          if (error) throw error.message;
+        }
       );
-    case ClientType.PartitionedQueueWithSessions:
-      return namespace.createQueueClient(
-        process.env.QUEUE_NAME_SESSION || "partitioned-queue-sessions"
+    })
+    .catch((err) => {
+      throw err.message;
+    });
+}
+
+async function recreateTopic(topicName: string, parameters: SBTopic): Promise<void> {
+  const env = getEnvVars();
+  await msRestNodeAuth
+    .loginWithServicePrincipalSecret(env.clientId, env.secret, env.tenantId)
+    .then(async (creds) => {
+      const client = await new ServiceBusManagementClient(creds, env.subscriptionId);
+      await client.topics.deleteMethod(
+        env.resourceGroup,
+        env.servicebusNamespace,
+        topicName,
+        function(error: any): void {
+          if (error) throw error.message;
+        }
       );
-    case ClientType.PartitionedSubscriptionWithSessions:
-      return namespace.createSubscriptionClient(
-        process.env.TOPIC_NAME_SESSION || "partitioned-topic-sessions",
-        process.env.SUBSCRIPTION_NAME_SESSION || "partitioned-topic-sessions-subscription"
+      await client.topics.createOrUpdate(
+        env.resourceGroup,
+        env.servicebusNamespace,
+        topicName,
+        parameters,
+        function(error: any): void {
+          if (error) throw error.message;
+        }
       );
-    case ClientType.UnpartitionedQueueWithSessions:
-      return namespace.createQueueClient(
-        process.env.QUEUE_NAME_NO_PARTITION_SESSION || "unpartitioned-queue-sessions"
+    })
+    .catch((err) => {
+      throw err.message;
+    });
+}
+
+async function recreateSubscription(
+  topicName: string,
+  subscriptionName: string,
+  parameters: SBSubscription
+): Promise<void> {
+  const env = getEnvVars();
+  await msRestNodeAuth
+    .loginWithServicePrincipalSecret(env.clientId, env.secret, env.tenantId)
+    .then(async (creds) => {
+      const client = await new ServiceBusManagementClient(creds, env.subscriptionId);
+      /*
+        Unlike Queues/Topics, there is no need to delete the subscription because
+        `recreateTopic` is called before `recreateSubscription` which would
+        delete the topic and the subscriptions before creating a new topic.
+      */
+      await client.subscriptions.createOrUpdate(
+        env.resourceGroup,
+        env.servicebusNamespace,
+        topicName,
+        subscriptionName,
+        parameters,
+        function(error: any): void {
+          if (error) {
+            throw error.message;
+          }
+        }
       );
-    case ClientType.UnpartitionedSubscriptionWithSessions:
-      return namespace.createSubscriptionClient(
-        process.env.TOPIC_NAME_NO_PARTITION_SESSION || "unpartitioned-topic-sessions",
+    });
+}
+
+export async function getSenderReceiverClients(
```

Comment:
Is there a reason why we had moved to using getSenderReceiverClients()?
The pro with previous getSender() getReceiver() was that we didn't have to make any changes if were to work with multiple senders/receivers

Reply:
Most of the tests involve both the sender and receiver.
In the beforeEach() section, we create both sender and receiver clients in all the test-cases.

In case of queues, we are deleting and creating the queue twice(in both the getSenderClient and getReceiverClient).
Same with subscriptions - deleting and creating the topic twice and then creating the subscription.

Since we are anyway creating both the sender and receiver clients in the beforeEach() section of every test, it makes sense to clean the service-bus entity only once.

Is there any case in our tests which creates only sender or receiver in the beforeeach() and not both ?

Reply:
We are using both, I was wondering about cases (which we actually haven't added yet) but where we create say two receivers with one sender to test some corner cases. Not sure if that's easy or intuitive to do so with 'getSenderReceiverClients()'

Reply:
When we have such cases, we can export `recreateTopic`/`recreateQueue` etc so they can be used directly

Reply:
Currently we don't have any tests where we create two senderClients(or receiverClients) and test them. 
Are we going to add such tests ?
For example, sending with senders from two different senderClients, receiving with receivers from two different receiverClients and operations on messages received from such receivers ?

Reply:
I think we will/should have those tests in eventually as we already are running into such scenarios.
I think combining this into getSenderReceiver() is unnecessary and not too convinced by the cleanup value it adds, but seems like there's an issue where we can't close a client without having to close the namespace. (Logged an issue for this #1192)
I'm okay to just get things working for now and look into this in a different issue.